src/Compilers/VisualBasic/Portable/BoundTree/BoundConversion.vb

```diff
@@ -78,6 +43,24 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Else
                 Debug.Assert(((ConversionKind And VisualBasic.ConversionKind.UserDefined) <> 0) = (Operand.Kind = BoundKind.UserDefinedConversion))
 
+                If Conversions.IsIdentityConversion(ConversionKind) Then
+                    Debug.Assert(ExtendedInfoOpt Is Nothing)
+                End If
```

Comment:
Consider moving `Conversions.IsIdentityConversion(ConversionKind)` inside the `Debug.Assert()`, to avoid the call in non-DEBUG build.